src/ops/ops.go

```diff
@@ -124,6 +125,71 @@ func (o *ops) SystemctlAction(action string, args ...string) error {
 	return errors.Wrapf(err, "Failed executing systemctl %s %s", action, args)
 }
 
+var ostreeOutputRegex = regexp.MustCompile(`Imported: (\w+)`)
+
+func (o *ops) importOSTreeCommit(liveLogger io.Writer) (string, error) {
+	ostreeReleasePullSpec := fmt.Sprintf("ostree-unverified-registry:%s", o.installerConfig.CoreosImage)
+	out, err := o.ExecPrivilegeCommand(liveLogger, "ostree", "container", "unencapsulate", "--authfile", "/root/.docker/config.json", "--quiet", "--repo", "/ostree/repo", ostreeReleasePullSpec)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to unencapsulate rhcos payload image: %s", out)
+	}
+
+	matches := ostreeOutputRegex.FindStringSubmatch(out)
+	if matches == nil {
+		return "", fmt.Errorf("got unexpected output from unencapsulate: \"%s\"", out)
+	}
+
+	return matches[1], nil
+}
+
+func (o *ops) WriteImageToExistingRoot(liveLogger io.Writer, ignitionPath string) error {
+	out, err := o.ExecPrivilegeCommand(liveLogger, "mount", "/sysroot", "-o", "remount,rw")
+	if err != nil {
+		return errors.Wrapf(err, "failed to remount sysroot: %s", out)
+	}
+	out, err = o.ExecPrivilegeCommand(liveLogger, "mount", "/boot", "-o", "remount,rw")
+	if err != nil {
+		return errors.Wrapf(err, "failed to remount boot: %s", out)
+	}
+
+	out, err = o.ExecPrivilegeCommand(liveLogger, "ostree", "admin", "stateroot-init", "install")
+	if err != nil {
+		return errors.Wrapf(err, "failed creating new stateroot: %s", out)
+	}
+
+	commit, err := o.importOSTreeCommit(liveLogger)
+	if err != nil {
+		return err
+	}
+	o.log.Infof("imported commit %s", commit)
+
+	out, err = o.ExecPrivilegeCommand(liveLogger, "ostree", "admin", "deploy",
+		"--stateroot", "install",
+		"--karg", "$ignition_firstboot",
+		"--karg", defaultIgnitionPlatformId,
```

Comment:
Shouldn't we pass platform ID based on the platform of the target image?

Reply:
What do you think for this one? We can always address bugs for this later if we run into anything.

[The other place that does this](https://github.com/openshift/assisted-installer/blob/ba4f15f38a7493893701bdd9c0e90ce5a41adb42/src/ops/ops.go#L953) is looking at the current kernel command line for platform to check, but I'm not sure we're really doing the same thing.